src/views/service-tree.coffee

```diff
@@ -181,19 +181,21 @@ define (require) ->
                 else
                     return ['service-node leaf']
 
-            listItems = @collection.filter((c) => c.get('unit_count') != 0).map (category) =>
-                selected = @selected(category.id)
-
-                rootId = category.get 'root'
-
-                id: category.get 'id'
-                name: category.getText 'name'
-                classes: classes(category).join " "
-                has_children: category.get('children').length > 0
-                unit_count: category.get('unit_count')
-                selected: selected
-                root_id: rootId
-                show_button_classes: @getShowButtonClasses selected, rootId
+            listItems = @collection
+                .filter((c) => c.get('unit_count') != 0 and (c.get('parent') != null or c.get('children').length > 0))
```

Comment:
Does this only filter on the categories in this case? I'm just wondering about the `children` check here, since all nodes will of course in the end have no children, but I'm guessing that the service points (which again, I'm guessing is what is showed last, without children) are not filtered here?

Reply:
This also filters the categories that doesn't have any children.  In case for categories not having children it checks if the parent is present on them.